sdk/appcenter/src/main/java/com/microsoft/appcenter/utils/context/AuthTokenContext.java

```diff
@@ -0,0 +1,111 @@
+package com.microsoft.appcenter.utils.context;
+
+import android.content.Context;
+import android.support.annotation.VisibleForTesting;
+
+import java.util.Collection;
+import java.util.LinkedHashSet;
+
+/**
+ * Utility to store and retrieve the latest authorization token.
+ */
+public class AuthTokenContext {
+
+    /**
+     * Unique instance.
+     */
+    private static AuthTokenContext sInstance;
+
+    /**
+     * Global listeners collection.
+     */
+    private final Collection<Listener> mListeners;
+
+    /**
+     * {@link ITokenStorage} instance that handles saving and encrypting token.
+     */
+    private ITokenStorage mTokenStorage;
+
+    /**
+     * Get unique instance.
+     *
+     * @return unique instance.
+     */
+    public static synchronized AuthTokenContext getInstance(Context context) {
+        if (sInstance == null) {
+            sInstance = new AuthTokenContext(context);
+        }
+        return sInstance;
+    }
+
+    /**
+     * A private constructor for the class.
+     *
+     * @param context application context instance.
+     */
+    private AuthTokenContext(Context context) {
+        mTokenStorage = TokenStorageFactory.getTokenStorage(context);
```

Comment:
Follow up to my comment below about moving this to Identity module: if we move the token storage from here into Identity, then identity can be the one to save the token in storage when there is a login. And if we want to be really really careful (which might not be necessary), Identity could even be a listener and update the storage when the auth token changes